scheduler/pkg/store/pipeline/store.go

```diff
@@ -97,6 +99,15 @@ func (ps *PipelineStore) InitialiseOrRestoreDB(path string) error {
 	if err != nil {
 		return err
 	}
+
+	go func() {
+		ticker := time.NewTicker(utils.DeletedResourceCleanupFrequency)
+		defer ticker.Stop()
```

Comment:
the for loop will run indefinitely and defer will not be called in practice.  Could consider a `done` channel pattern as well, that can be linked to the process exit.  Otherwise just remove?

Reply:
I removed it - probably no need to shut it down.